Changing OnUserQuit to allow quit message modify.

iwalkalone69 opened the pull request:
The purpose of this pull request is to allow modifying quit message like we can do with part message in OnUserPart event.

The change is just removing const from parameter to allow its modification in modules.

SaberUK commented:
It would be better if an `OnUserPreQuit` event was to be introduced which is called with FIRST_MOD_RESULT. That way modules which just want to see the quit don't have to worry if a module after them in the chain has changed the message.

iwalkalone69 commented:
Thanks.

I will add this event and revert the change on the existing one according to you comment.

iwalkalone69 commented:
I've reverted changes on **OnUserQuit** event and added **OnUserPreQuit** event.

Hope it is done right. If not, tell me how should be done. Thanks.

SaberUK commented on include/modules.h:
```diff
@@ -304,6 +304,16 @@ class CoreExport Module : public classbase, public usecountbase
+	 * @param message The user's quit message (as seen by non-opers)
+	 * @param oper_message The user's quit message (as seen by opers)
+	 */
+	virtual ModResult OnUserPreQuit(User* user, std::string &message);
```
This should take a LocalUser not a User.

Also, the & should be against the type not against the variable name.

SaberUK commented on src/usermanager.cpp:
```diff
@@ -174,16 +174,23 @@ void UserManager::QuitUser(User* user, const std::string& quitreason, const std:
+
+	ModResult MOD_RESULT;
+	FIRST_MOD_RESULT(OnUserPreQuit, MOD_RESULT, (user, reason));
+	if (MOD_RESULT == MOD_RES_DENY) {
```
Thereis no need to have {} on this.

iwalkalone69 commented:
Applied change suggested in the code commented.

iwalkalone69 commented:
Is everything ok now? Thanks.

SaberUK commented:
Hi, i'll try to get around to merging this before the last alpha.

iwalkalone69 commented:
Rebased in insp3 branch (PR #1629).
